packages/viem/package.json

```diff
@@ -14,13 +14,15 @@
     "typecheck": "tsc --noEmit --emitDeclarationOnly false",
     "typecheck:ci": "tsc --noEmit --emitDeclarationOnly false",
     "build": "tsc && resolve-tspaths",
+    "generate": "pnpm dlx tsx ./scripts/generate.ts",
```

Comment:
nit: but would it be helpful to have a `fetchandgenerate` script as well that also runs the `fetch:artifacts` script since this is dependent on that?

Reply:
This will also go in the direction of happening in CI as this evolves so we don't need to worry about running anything locally.

Reply:
@tremarkley updated on this commit https://github.com/ethereum-optimism/ecosystem/pull/479/commits/32dd1210906a21dc59e63cbce4fe5f233f5b6205

This makes `fetch:artifacts` automatically happen when we attempt to run a generate command in a cacheable way. If we have `N` packages that all need to call `fetch:artifacts` it will only happen once and the others will read from the cache